File: app/src/main/java/com/zhihudailytest/CustomUI/MyLineayLayout.java
package com.zhihudailytest.CustomUI;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.LinearLayout;
import android.widget.ListView;

/**
 * Created by Administrator on 2016/8/3.
 */

public class MyLineayLayout extends LinearLayout {
    public MyLineayLayout(Context context) {
        super(context);
    }

    public MyLineayLayout(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public MyLineayLayout(Context context, AttributeSet attrs,
                                 int defStyle) {
        super(context, attrs, defStyle);
    }

    @Override
    /**
     * 重写该方法，达到使ListView适应ScrollView的效果
     */
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        int expandSpec = MeasureSpec.makeMeasureSpec(Integer.MAX_VALUE >> 2,
                MeasureSpec.AT_MOST);
        super.onMeasure(widthMeasureSpec, expandSpec);
    }
}

File: app/src/main/java/com/zhihudailytest/CustomUI/MyLineayLayout.kt
package com.zhihudailytest.CustomUI

import android.content.Context
import android.util.AttributeSet
import android.view.View
import android.widget.LinearLayout

/**
 * Created by Administrator on 2016/8/3.
 */

class MyLineayLayout : LinearLayout {
    constructor(context: Context) : super(context) {}

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {}

    constructor(context: Context, attrs: AttributeSet,
                defStyle: Int) : super(context, attrs, defStyle) {
    }

    override
            /**
             * 重写该方法，达到使ListView适应ScrollView的效果
             */
    fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        val expandSpec = View.MeasureSpec.makeMeasureSpec(Integer.MAX_VALUE shr 2,
                View.MeasureSpec.AT_MOST)
        super.onMeasure(widthMeasureSpec, expandSpec)
    }
}